### PR TITLE
BugFix: a fix for java.util.NoSuchElementException: None.get

### PR DESCRIPTION
**Pull request by nikolayilyin**
to fix a fatal exception:

```
09:05:47.099 [main] INFO  beam.sim.BeamMobsim - Filling in secondary trips in plans
09:09:34.884 [main] ERROR o.m.c.controler.AbstractController - Mobsim did not complete normally! afterMobsimListeners will be called anyway.
java.util.NoSuchElementException: None.get
	at scala.None$.get(Option.scala:529)
	at scala.None$.get(Option.scala:527)
	at beam.replanning.SupplementaryTripGenerator.generateSubtour(SupplementaryTripGenerator.scala:163)
	at beam.replanning.SupplementaryTripGenerator.$anonfun$generateNewPlans$1(SupplementaryTripGenerator.scala:82)
	at beam.replanning.SupplementaryTripGenerator.$anonfun$generateNewPlans$1$adapted(SupplementaryTripGenerator.scala:78)
```

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/2864)

### CODE REVIEW

**Comment by nikolayilyin** [comment]
test!

File: src/main/scala/beam/replanning/SupplementaryTripGenerator.scala [hunk 1]
```diff
@@ -160,12 +160,9 @@ class SupplementaryTripGenerator(
...
 
-        val makeTrip: Boolean = tripMNL.sampleAlternative(tripChoice, r).get.alternativeType
-
-        if (makeTrip) {
-          destinationMNL.sampleAlternative(modeChoice, r)
-        } else {
-          None
+        tripMNL.sampleAlternative(tripChoice, r) match {
+          case Some(mnlSample) if mnlSample.alternativeType => destinationMNL.sampleAlternative(modeChoice, r)
+          case _                                            => None
```
Comment (JustinPihony) [inline src/main/scala/beam/replanning/SupplementaryTripGenerator.scala:165]: How about a collect?

```suggestion
        tripMNL.sampleAlternative(tripChoice, r) collect {
          case mnlSample if mnlSample.alternativeType => destinationMNL.sampleAlternative(modeChoice, r)
```

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

File: src/main/scala/beam/replanning/SupplementaryTripGenerator.scala (hunk 1 above)
Comment (JustinPihony) [reply src/main/scala/beam/replanning/SupplementaryTripGenerator.scala:165]: Sorry, I didn't pay attention to the final type - change my collect to a map and that would have been all that was needed.

**Pull request merged**